File: java/Diretor.java
public class Diretor {
    private Builder builder;

    public Diretor(Builder builder) {
        this.builder = builder;
    }

    public Bolo buildBoloPequenoSemCobertura() {
        return builder.adicionarRecheio("Creme de baunilha").adicionarTamanho("Pequeno").getBolo();
    }
    public Bolo buildBoloGrande () {
        return builder.adicionarCobertura("Morango").adicionarRecheio("Chantilly").adicionarTamanho("Grande").getBolo();
    }
}

File: php/Diretor.php
<?php

class Diretor
{
    private $builder;

    public function setBuilder(BuilderBolo $builder): void
    {
        $this->builder = $builder;
    }

    public function buildBoloPequenoSemRecheio(): void
    {
        $this->builder->adicionarCobertura("Chocolate");
        $this->builder->adicionarTamanho("Pequeno");
    }

    public function buildBoloGrande()
    {
        $this->builder->adicionarRecheio("Morango");
        $this->builder->adicionarCobertura("Abacaxi");
        $this->builder->adicionarTamanho("Grande");
    }


}





?>